config/init/common/lxc-net.in

```diff
@@ -51,6 +55,52 @@ _ifup() {
     ip link set dev ${LXC_BRIDGE} up
 }
 
+start_ipv6() {
+    LXC_IPV6_ARG=""
+    if [ -n "$LXC_IPV6_ADDR" ] && [ -n "$LXC_IPV6_MASK" ] && [ -n "$LXC_IPV6_NETWORK" ]; then
+        echo 1 > /proc/sys/net/ipv6/conf/all/forwarding
+        echo 0 > /proc/sys/net/ipv6/conf/${LXC_BRIDGE}/autoconf
+        ip -6 addr add dev ${LXC_BRIDGE} ${LXC_IPV6_ADDR}/${LXC_IPV6_MASK}
+        LXC_IPV6_ARG="--dhcp-range=${LXC_IPV6_ADDR},ra-only --listen-address ${LXC_IPV6_ADDR}"
+    fi
+}
+
+start_iptables() {
+    start_ipv6
+    if [ -n "$LXC_IPV6_ARG" ] && [ "$LXC_IPV6_NAT" = "true" ]; then
+        ip6tables $use_iptables_lock -t nat -A POSTROUTING -s ${LXC_IPV6_NETWORK} ! -d ${LXC_IPV6_NETWORK} -j MASQUERADE
+    fi
+    iptables $use_iptables_lock -I INPUT -i ${LXC_BRIDGE} -p udp --dport 67 -j ACCEPT
+    iptables $use_iptables_lock -I INPUT -i ${LXC_BRIDGE} -p tcp --dport 67 -j ACCEPT
+    iptables $use_iptables_lock -I INPUT -i ${LXC_BRIDGE} -p udp --dport 53 -j ACCEPT
+    iptables $use_iptables_lock -I INPUT -i ${LXC_BRIDGE} -p tcp --dport 53 -j ACCEPT
+    iptables $use_iptables_lock -I FORWARD -i ${LXC_BRIDGE} -j ACCEPT
+    iptables $use_iptables_lock -I FORWARD -o ${LXC_BRIDGE} -j ACCEPT
+    iptables $use_iptables_lock -t nat -A POSTROUTING -s ${LXC_NETWORK} ! -d ${LXC_NETWORK} -j MASQUERADE
+    iptables $use_iptables_lock -t mangle -A POSTROUTING -o ${LXC_BRIDGE} -p udp -m udp --dport 68 -j CHECKSUM --checksum-fill
+}
+
+start_nftables() {
+    start_ipv6
+    # It is not possible to delete rules in nftables using directly the same
```

Comment:
nftables doesn't have any built in tables & chains like iptables does. However because there is an iptables shim that populates nftables, it can sometimes look like nftables has the the `filter` table already.

So for example, on Ubuntu 20.04, running:

```
sudo nft list ruleset
```

Produces no output, and running:

```
sudo nft add chain ip6 nat lxc-nat
```

Fails with:

```
Error: No such file or directory
add chain inet filter lxc-filter
```

This is because the inet filter table doesn't exist.

My recommendation is that you test this script by resetting the nftables state first using `sudo nft flush ruleset`.

Reply:
Also on that point, is there any reason you aren't using a separate `lxc` inet table, and then setting up the non-namespaced chains inside that table (with netfilter hooks), rather than using the `filter` table?

This would potentially make removing the rules even easier as you can just delete the whole table.

The nice thing is that you can use the `add` command and if the entity already exists it won't error, so that allows you to always `add` the table and chains you need, and if they don't exist they'll get setup.

I've used this effect when I wanted to ensure that the chains exist in order to flush their rules in a single call to nftables:

https://github.com/lxc/lxd/blob/master/lxd/firewall/drivers/drivers_nftables_templates.go#L101-L109

Reply:
Thank you very much for the comments. You're totally right, I assumed that the `nat` and `filter` tables existed, my bad. And had not thought about creating a separate table, which definitely makes total sense. I'll fix that up.